src/views/Pools/components/CakeVaultCard/RecentCakeProfitRow.tsx

```diff
@@ -1,26 +1,44 @@
 import React from 'react'
-import BigNumber from 'bignumber.js'
 import { Flex, Text } from '@pancakeswap/uikit'
-import { useTranslation } from 'contexts/Localization'
 import { useWeb3React } from '@web3-react/core'
+import { useTranslation } from 'contexts/Localization'
+import { useGetApiPrice } from 'state/hooks'
+import useGetVaultUserInfo from 'hooks/cakeVault/useGetVaultUserInfo'
+import useGetVaultPublicData from 'hooks/cakeVault/useGetVaultPublicData'
+import { getBalanceNumber } from 'utils/formatBalance'
+import { getCakeAddress } from 'utils/addressHelpers'
+import { convertSharesToCake } from 'views/Pools/helpers'
 import RecentCakeProfitBalance from './RecentCakeProfitBalance'
 
-interface RecentCakeProfitRowProps {
-  cakeAtLastUserAction: BigNumber
-  userShares: BigNumber
-}
-
-const RecentCakeProfitCountdownRow: React.FC<RecentCakeProfitRowProps> = ({ cakeAtLastUserAction, userShares }) => {
+const RecentCakeProfitCountdownRow = () => {
   const { t } = useTranslation()
   const { account } = useWeb3React()
+  const { cakeAtLastUserAction, userShares, lastUserActionTime } = useGetVaultUserInfo()
   const shouldDisplayCakeProfit =
     account && cakeAtLastUserAction && cakeAtLastUserAction.gt(0) && userShares && userShares.gt(0)
 
+  const { pricePerFullShare } = useGetVaultPublicData()
+  const currentSharesAsCake = convertSharesToCake(userShares, pricePerFullShare)
+  const cakeProfit = currentSharesAsCake.cakeAsBigNumber.minus(cakeAtLastUserAction)
+  const cakeToDisplay = cakeProfit.gte(0) ? getBalanceNumber(cakeProfit, 18) : 0
+
+  const cakePrice = useGetApiPrice(getCakeAddress())
+  const dollarValueOfCake = cakeProfit.times(cakePrice)
+  const dollarValueToDisplay = dollarValueOfCake.gte(0) ? getBalanceNumber(dollarValueOfCake, 18) : 0
```

Comment:
A lot of this logic isn't new, just moved out of the child `<RecentCakeProfitBalance>` component